File: Lista01/PY/ex06.py
# Definindo a classe Motor
class Motor:
    def __init__(self, cilindrada, potencia):
        self.cilindrada = cilindrada
        self.potencia = potencia

    # Método para exibir os detalhes do motor
    def exibir_detalhes_motor(self):
        print(f"Motor: {self.cilindrada} cilindradas, {self.potencia} HP")

# Definindo a classe Carro
class Carro:
    def __init__(self, marca, modelo, motor):
        self.marca = marca
        self.modelo = modelo
        self.motor = motor  # Composição: Carro contém um objeto Motor

    # Método para exibir os detalhes do carro e do motor
    def exibir_detalhes(self):
        print(f"Carro: {self.marca} {self.modelo}")
        self.motor.exibir_detalhes_motor()  # Chamando método da classe Motor

# Exemplo de uso com Fiat Pulse
motor_pulse = Motor(1000, 130)  # Criando um objeto Motor (1.0 turbo, 130 HP)
carro_pulse = Carro("Fiat", "Pulse", motor_pulse)  # Associando o motor ao carro

# Exibindo os detalhes do carro e do motor
carro_pulse.exibir_detalhes()

File: Lista01/GO/ex06.go
package main

import "fmt"

// Definindo a struct Motor
type Motor struct {
	potencia int
}

// Função associada ao Motor para exibir a potência
func (m Motor) ExibirPotencia() {
	fmt.Printf("Potência do motor: %d HP\n", m.potencia)
}

// Definindo a struct Carro que possui um Motor
type Carro struct {
	marca  string
	modelo string
	motor  Motor
}

// Função para exibir detalhes do carro e do motor
func (c Carro) ExibirDetalhes() {
	fmt.Printf("Carro: %s %s\n", c.marca, c.modelo)
	c.motor.ExibirPotencia()
}

func main() {
	motor := Motor{potencia: 150}
	carro := Carro{marca: "Fiat", modelo: "Pulse", motor: motor}

	carro.ExibirDetalhes()
}
